### DISCUSSION TITLE
How to use the Google One Tap solution in your webapp to authenticate users

### DISCUSSION THREAD

**Discussion by giscus** [piraces/blog#264] [Announcements]
# How to use the Google One Tap solution in your webapp to authenticate users

Google Identity services provides us of seamless sign-in and sign-up flows (among other things), that we can use to ease the management of user authentication and sign-in to our website. This is the case of "Google One Tap", which I will explain in this post.

https://piraces.dev/posts/how-to-use-google-one-tap/

**Comment by iBasit** [comment]
The token is valid for 1hr.

on server side validation of the token after the 1st hr, it will show invalid token.

how do we refresh token within 1hr, so we don't have to prompt user again to fetch the valid token?

**Reply by piraces** [reply]
Please take a look at [this guide](https://developers.google.com/identity/gsi/web/guides/migration?hl=en#session_state_and_listeners). It appears that the session is handled by Google and no refresh token is available... Apparently, there is no programmatic flow to refresh the token.

**Comment by thanhtutzaw** [comment]
When I refresh I am not sign in anymore . Meaning after 2nd refresh user need to login again . I don't think it is normal .

**Comment by dmoritz10** [comment]
Where are the discoveryDocs and scope entered ?  Currently, I use:

await gapi.client.init({
                apiKey:                 signin.API_KEY,
                clientId:               signin.CLIENT_ID,
                discoveryDocs:          signin.DISCOVERY_DOCS,
                fetch_basic_profile:    true,
                scope:                  signin.SCOPES

            }).then(function () { ...

**Reply by dmoritz10** [reply]
I see now that Authentication  and Authorization have been seperated.  I don't suppose you've don't a similar post on Authorization :)

**Comment by frknbasaran** [comment]
Thanks for this article. Even official google documentation is not clear as this one. :)
